st3/sublime_lib/syntax.py

```diff
@@ -15,57 +12,58 @@
 SyntaxInfo.__new__.__defaults__ = (None, None, False)  # type: ignore
 
 
-def _parse_yaml_value(value):
-    if value.startswith("'"):
-        return value[1:-1].replace("''", "'")
-    elif value.startswith('"'):
-        # JSON and YAML quotation rules are very similar, if not identical
-        return sublime.decode_value(value)
-    elif value == "true":
-        return True
-    elif value == "false":
-        return False
-    elif value == "null":
-        return None
-    else:
-        # Does not handle numbers because we don't expect any
-        return value
-
-
-def get_yaml_metadata(text):
-    return projection(
-        dict(
-            map(_parse_yaml_value, match.groups())
-            for match in re.finditer(r'(?m)^(\S.*?):\s*(.*)\s*$', text)
-        ),
-        {'name', 'hidden', 'scope'}
-    )
+def get_sublime_syntax_metadata(path):
+    yaml = parse_simple_top_level_keys(path.read_text())
+    return {
+        'name': yaml.get('name') or path.stem,
+        'hidden': yaml.get('hidden', False),
+        'scope': yaml.get('scope'),
+    }
+
+
+def get_tmlanguage_metadata(path):
+    tree = plistlib.readPlistFromBytes(path.read_bytes())
+
+    return {
+        'name': tree.get('name') or path.stem,
+        'hidden': tree.get('hidden', False),
+        'scope': tree.get('scopeName'),
+    }
+
+
+def get_hidden_tmlanguage_metadata(path):
+    tree = plistlib.readPlistFromBytes(path.read_bytes())
 
+    return {
+        'name': path.stem, # `name` key is ignored
+        'hidden': True, # `hidden` key is ignored
+        'scope': tree.get('scopeName'),
+    }
 
-def get_xml_metadata(text):
-    tree = plistlib.readPlistFromBytes(text)
 
-    return projection(tree, {
-        'name': 'name',
-        'hidden': 'hidden',
-        'scopeName': 'scope',
-    })
+SYNTAX_TYPES = {
+    '.sublime-syntax': get_sublime_syntax_metadata,
+    '.tmLanguage': get_tmlanguage_metadata,
+    '.hidden-tmLanguage': get_hidden_tmlanguage_metadata,
+}
 
 
 def get_syntax_metadata(path):
-    if path.suffix == '.sublime-syntax':
-        meta = get_yaml_metadata(path.read_text())
-    elif path.suffix == '.tmLanguage':
-        meta = get_xml_metadata(path.read_bytes())
-    else:
-        raise TypeError("%s is not a syntax definition." % path)
+    return SyntaxInfo(
+        path=str(path),
+        **SYNTAX_TYPES[path.suffix](path)
+    )
+
 
-    return SyntaxInfo(path=str(path), **meta)
+def shadowed(path):
+    return (
+        path.suffix in {'.tmLanguage', '.hidden-tmLanguage'}
+        and path.with_suffix('.sublime-syntax').exists()
```

Comment:
We could save this call (that ends up making another `find_resources` call for every non-sublime-syntax syntax) by caching the initial `ResourcePath.glob_resources('')` call in line 85 and checking whether that contains the sublime-syntax-suffixed path.

I'd probably inine this function into `list_syntaxes` after `glob_resources` was assigned to a variable.

Reply:
Done.